ament_copyright/ament_copyright/parser.py

```diff
@@ -116,10 +116,17 @@ def parse(self):
         # get first comment block without leading comment tokens
         block, _ = get_comment_block(self.content, index)
         if not block:
```

Comment:
Will the return here cause the parser to fail in the case where only a multiline comment block exists in the entire file? (With no # includes)

e.g.

```
/*
 * Copyright 2018 Open Source Robotics Foundation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
```

Reply:
Yes, thanks for pointing this out! This would fail for a file with only multiline comments and with no `//` comments or no `#include` in it. So most `C++` sources would probably return something for the first call to `get_comment_block` but I agree that there is an edge case.

I was able to simplify some things by making `search_copyright_information` check for `None` and I do strict checking per your comment https://github.com/ament/ament_lint/pull/361#issuecomment-1157969641